### DISCUSSION TITLE
github pages gives  DNS_PROBE_FINISHED_NXDOMAIN

### DISCUSSION THREAD

**Discussion by xkeshav** [community/community#81851] [Actions] [answered]
### Select Topic Area

General

### Body

I have public repo which have custom domain and build and deploy is succesfull and it say deployed successfully to custom domain but when I open the domain it says SITE cant be reached : DNS_PROBE_FINISHED_NXDOMAIN

I have verified the domain and added custom TX record as per github instructions  and can see **verified** on  https://github.com/settings/pages

my domain is from google 

under DNS name server tab it says your domain using google domain server ( which are read only)

so what is the issue here?

even on github pages whe it check for DNS it faill with error InvalidDNSError

Both www.[domainname].com and its alternate name are improperly configured
Domain's DNS record could not be retrieved. For more information, see [documentation](https://docs.github.com/articles/setting-up-a-custom-domain-with-github-pages/) (InvalidDNSError).

**Comment by hiraksarkar** [comment]
Same problem for me. Is there an update?

**Reply by xkeshav** [reply]
Yes, I have resolved it

add CNAME entry in domain DNS with value of 

`<your githubusername>.github.io`

**Comment by xkeshav** [comment, marked as answer]
Add one more entry in DNS tab in google domain



www | CNAME | xkeshav.github.io

note: change *xkeshav* with your own github username

Make sure you have A entry also in DNS server

Which are described in github pages ( 4 IP in value field) keep name empty

**Comment by Martichka9** [comment]
I have the same problem but my domain is from another company. I've already asked to add the TXT record mentioned in the docs, also just asked the CNAME records to be added. If nothing changes - do you have an idea what else could be the problem?
Also can you share more info about the CNAME records that should be included in the repo?

**Reply by xkeshav** [reply]
Do below 

- Add a file name **CNAME**  in your repo main branch and content of CNAME will be just your domain name without _http_ and _www_,  you can see [my repo](https://github.com/xkeshav/threadzip/blob/main/CNAME) for the same.
- Create an empty `.nojekyll` file in the root folder to turn off Jekyll.

**Reply by Martichka9** [reply]
Thank you! I see that in the manuals there are more then one records that need to be added to the DNS. For example the TXT one, but if I add CNAME record I'm unable to add other records (some kind of restriction of the provider). 
The TXT record is mandatory right?

**Reply by xkeshav** [reply]
yes. TXT record in mandatory; which is the provider btw?

**Reply by Martichka9** [reply]
Superhosting in Bulgaria. A friend of mine recommended it to me, but I'm totally not satisfied at all so far.

**Reply by Martichka9** [reply]
If anyone have the same issue and is unable to add the CNAME along with the TXT or similar:

I've removed all records in the DNS created so far and added the 4 IPs given in the GH Pages manual for custom domains. 
Along them added the TXT needed for validation. This is the one given in YourProfile -> Settings -> Pages -> Add a domain.
Yesterday already added the CNAME file to my repo. Not sure if it is still needed. Will test this later and update the comment if needed.
Waited the DNS propagation to happen, then added the apex domain to my repo from GH Pages repo -> Settings -> Pages -> Custom domain.
After about 5-10 mins the webpage is properly loading with the desired custom domain name. 

Hope this helps someone with the same issue with the providers.

**Reply by Martichka9** [reply]
Thank you very much for assistance Keshav!
